RFCs/0014-TestCase-FullyQualifiedName.md

```diff
@@ -0,0 +1,48 @@
+# 0014 Guidelines for setting the FullyQualifiedName on TestCase
+
+## Summary
+This note clarifies the guidelines that test adapters should consider when setting the FullyQualifiedName on test cases before returning them from the discovery process.
+
+## Overview
+There are a broad variety of choices that current Test Adapter implementers have made when setting the FullyQualifiedName on TestCase objects created during test discovery. This note lays out some basic guidelines to make the usage of FullyQualifiedName more useful across adapters.
+
+## Usage of FullyQualifiedName
+The Test Platform uses FullyQualifiedName as a string moniker to name tests during discovery. Visual Studio may filter tests by FullyQualifiedName in order to more quickly filter and run specific tests. The guidelines below clarify the usage of FullyQualifiedName so that these optimized test execution scenarios can be realized.
+
+## Guidelines
+1. **FullyQualifiedName does NOT need to be unique across your test set.** Internally, TestCase objects are tracked by a unique ID (GUID) which can either be set by the Test Adapter implementor, or automatically generated otherwise. The FullyQualifiedName should be unique if possible, but it's not a requirement.
+1. **FullyQualifiedName should be deterministic** FullyQualifiedName should have the same value for the same test across different discovery sessions. This means that including random data, time-dependent data or environment specific data in the FullyQualifiedName should be avoided.
+1. **FullyQualifiedName is expected to be a reference to the method that implements the test** The Test Platform is expecting the FullyQualifiedName to refer to a particular test case. If there are multiple tests that are instantiated from a test method (data-driven tests for instance), then the FullyQualifiedName should distinguish these using parentheses to represent data parameters and angle brackets to represent type parameters. See below.
+
+## Recommended Form
+FullyQualifiedName is expected to be a reference to the TestCase method that implements the test (or tests). The FullyQualifiedName should be unique to a particular TestCase instance, if possible.
+
+The recommended forms of FullyQualifiedName are described by the following examples:
```

Comment:
I think this form only applies to languages with NS.Class.Method concepts. Normalization will be skipped for languages that don't have these (e.g. Python or JS)?

Reply:
That's a great point. Normalization is only used for source-based test discovery. I'll add a comment clarifying that.